src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -2198,3 +2198,39 @@ class LockTableStatementSegment(BaseSegment):
             Ref("TableReferenceSegment"),
         ),
     )
+
+
+class TableExpressionSegment(BaseSegment):
+    """The main table expression e.g. within a FROM clause.
+
+    Override to add Object unpivoting.
+    """
+
+    type = "table_expression"
+    match_grammar: Matchable = OneOf(
```

Comment:
Since we're only adding one thing to an existing pretty large list from the parent, should we use the copy/insert technique instead?

Reply:
Good shout. Will fix that.

Reply:
Done!

Reply:
Gentle nudge for re-review @barrywhart